wls-ergebnismeldung-service/src/test/java/de/muenchen/oss/wahllokalsystem/ergebnismeldungservice/rest/ausdruck/AusdruckControllerTest.java

```diff
@@ -0,0 +1,139 @@
+        }
+
+        @Test
+        void should_returnNullWithHttpStatusNotFound_when_serviceReturnsNoData() {
+            val wahlID = "wahlID";
+            val wahlbezirkID = "wahlbezirkID";
+            val meldungsArt = Meldungsart.V1;
+
+            when(ausdruckService.getAusdruck(any())).thenReturn(null);
+            when(ausdruckReadDTOMapper.toDTO(any())).thenReturn(null);
+
+            val result = unitUnderTest.getAusdruck(wahlID, wahlbezirkID, meldungsArt);
+
+            Assertions.assertThat(result.getBody()).isNull();
+            Assertions.assertThat(result.getStatusCode()).isEqualTo(HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @Nested
+    class PostAusdruck {
+
+        @Test
+        void should_callServiceWithModel_when_calledWithData() {
+            var clock = Clock.fixed(Instant.now(), ZoneOffset.UTC);
+            var mockedInstant = Instant.now(clock);
+            MockedStatic<Instant> mockedStatic = mockStatic(Instant.class, Mockito.CALLS_REAL_METHODS);
+            mockedStatic.when(Instant::now).thenReturn(mockedInstant);
+
+            val wahlID = "wahlID";
+            val wahlbezirkID = "wahlbezirkID";
+            val meldungsArt = Meldungsart.V1;
+            val content = "Testcontent";
+            val erstelltAm = Instant.now();
+            val id = new WahlUndBezirkIDUndMeldungsart(wahlbezirkID, wahlID, meldungsArt);
+            val ausdruckWriteDTO = new AusdruckWriteDTO(content);
+
+            val mockedAusdruckModel = new AusdruckModel(id, content, erstelltAm);
+            when(ausdruckWriteDTOMapper.toModel(ausdruckWriteDTO, id, erstelltAm)).thenReturn(mockedAusdruckModel);
+
+            unitUnderTest.postAusdruck(wahlID, wahlbezirkID, meldungsArt, ausdruckWriteDTO);
+
+            Mockito.verify(ausdruckService).saveAusdruck((mockedAusdruckModel));
+
+            mockedStatic.close();
+        }
+
+    }
+
+    @Nested
+    class GetAllAusdrucke {
+
+        @Test
+        void should_returnHttpStatusOkAndData_when_dataIsFound() {
+            val wahlID = "wahlID";
+            val wahlbezirkID = "wahlbezirkID";
+            val content = "Testcontent";
+            val erstelltAm = Instant.now();
+
+            val mockedServiceModel = List.of(new AusdruckModel(new WahlUndBezirkIDUndMeldungsart(wahlbezirkID, wahlID, Meldungsart.V1), content, erstelltAm),
+                    new AusdruckModel(new WahlUndBezirkIDUndMeldungsart(wahlbezirkID, wahlID, Meldungsart.V3), content, erstelltAm));
+            val mockedMappedServiceDTO = List.of(new AusdruckReadDTO(wahlbezirkID, wahlID, Meldungsart.V1, content, erstelltAm),
+                    new AusdruckReadDTO(wahlbezirkID, wahlID, Meldungsart.V3, content, erstelltAm));
+
+            Mockito.when(ausdruckService.getAllAusdrucke(wahlID, wahlbezirkID)).thenReturn(mockedServiceModel);
+            Mockito.when(ausdruckReadDTOMapper.fromListOfAusdruckModelToListOfAusdruckReadDTO(mockedServiceModel)).thenReturn(mockedMappedServiceDTO);
```

Comment:
es sollte in der Klasse einheitlich sein. Das wir hier mit einem static-Import arbeiten stört mich wenig

```suggestion
            when(ausdruckService.getAllAusdrucke(wahlID, wahlbezirkID)).thenReturn(mockedServiceModel);
            when(ausdruckReadDTOMapper.fromListOfAusdruckModelToListOfAusdruckReadDTO(mockedServiceModel)).thenReturn(mockedMappedServiceDTO);
```

Reply:
Vorschlag abgelehnt - statt dessen static Import entfernt